Formula/synergy-core.rb

```diff
@@ -0,0 +1,274 @@
+class SynergyCore < Formula
+  desc "Synergy, the keyboard and mouse sharing tool"
+  homepage "https://symless.com/synergy"
+  url "https://github.com/symless/synergy-core/archive/refs/tags/1.14.3.5-stable.tar.gz"
+  sha256 "9259fc8aa02610121932cd5887c4d878beb08b2f7190e834c0584d53b36f5f9b"
+
+  # The synergy-core/LICENSE file contains the following preamble:
+  #    This program is released under the GPL with the additional exemption
+  #    that compiling, linking, and/or using OpenSSL is allowed.
+  # This preamble is followed by the text of the GPL-2.0.
+  #
+  # The list of SPDX license exceptions does not contain an exception corresponding
+  # to synergy-core's special OpenSSL exception. The openvpn-openssl-exception
+  # exception is very similar, but it is not the same, so it would be inappropriate
+  # to use it here.
+  #
+  # Unfortunately, writing an exception like this:
+  #     with: :cannot_represent
+  # causes `brew audit` to raise the following error:
+  #     Formula synergy-core contains invalid or deprecated SPDX license exceptions: [:cannot_represent].
+  # As a result, it appears we cannot mention the exception in the `license` definition.
+  license "GPL-2.0-only"
+
+  head "https://github.com/symless/synergy-core.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "openssl@3"
+  depends_on "pugixml"
+  depends_on "qt@5"
+
+  on_linux do
+    depends_on "gcc" => :build # For C++17 support.
+    depends_on "pkg-config" => :build
+    depends_on "gdk-pixbuf"
+    depends_on "glib"
+    depends_on "libnotify"
+    depends_on "libxkbfile"
+  end
+
+  fails_with gcc: "5" # Synergy requires C++17, which is not supported by gcc@5.
+
+  def install
+    # Use the standard brew installation path.
+    inreplace "CMakeLists.txt",
+              "set (SYNERGY_BUNDLE_DIR ${CMAKE_BINARY_DIR}/bundle)",
+              "set (SYNERGY_BUNDLE_DIR ${CMAKE_INSTALL_PREFIX}/bundle)"
+
+    # Use the standard brew location of the pugixml library.
+    #
+    # This `inreplace` block can be replaced with a CMake variable once
+    # upstream accepts the following patch:
+    #   https://github.com/symless/synergy-core/pull/7158
+    inreplace "src/lib/synergy/CMakeLists.txt" do |s|
+      # Remove references to nonstandard pugixml path.
+      s.gsub!("include_directories(\"../../../ext/pugixml/src\")\n" \
+              "list(APPEND arch_sources \"../../../ext/pugixml/src/pugixml.cpp\")", "")
+      # Insert reference to brew's pugixml path.
+      s.gsub!(/\s*target_link_libraries\((?<items>synlib arch .*)\)$/,
+              "\n find_package(PugiXML CONFIG REQUIRED)" \
+              "\n target_link_libraries(\\k<items> pugixml)\n")
+    end
```

Comment:
Can you patch the formula with that PR content for now? That makes is much easier to keep track of the patches

Reply:
Upstream has now merged a pull request to fix this issue. See symless/synergy-core#7119. We still need the patch until upstream releases a new stable version though.